### ISSUE TITLE
20 Practice Exercises

### ISSUE THREAD

**Issue by loziniak** [exercism/red#31]
At least 20 Practice Exercises are needed to launch track.

### checklist:

exercise name (so-called "slug") is linked to a **description**.
task's **difficulty** is in parentheses. the choice is arbitrary, based on personal reception of exercise description. it's open for discussion/change.
this list is chosen randomly from [Exercism's problem database](https://github.com/exercism/problem-specifications/tree/main/exercises).
it's **sorted** by difficulty, and this order should be kept also in `config.json`.

- [x] [hello-world](https://github.com/exercism/problem-specifications/blob/main/exercises/hello-world/description.md) (1) // #7
- [x] [two-fer](https://github.com/exercism/problem-specifications/blob/main/exercises/two-fer/description.md) (2) // b2c5112
- [x] [reverse-string](https://github.com/exercism/problem-specifications/blob/main/exercises/reverse-string/description.md) (1) // #44
- [x] [resistor-color](https://github.com/exercism/problem-specifications/blob/main/exercises/resistor-color/description.md) (2) // #45
- [x] [protein-translation](https://github.com/exercism/problem-specifications/blob/main/exercises/protein-translation/description.md) (3) // #47 
- [x] [space-age](https://github.com/exercism/problem-specifications/blob/main/exercises/space-age/description.md) (3) // 7f25736
- [x] [raindrops](https://github.com/exercism/problem-specifications/blob/main/exercises/raindrops/description.md) (3) // a71ee29
- [x] [resistor-color-duo](https://github.com/exercism/problem-specifications/blob/main/exercises/resistor-color-duo/description.md) (4) // #46
- [x] [rotational-cipher](https://github.com/exercism/problem-specifications/blob/main/exercises/rotational-cipher/description.md) (4) // 67970843
- [x] [darts](https://github.com/exercism/problem-specifications/blob/main/exercises/darts/description.md) (4) // #53
- [x] [largest-series-product](https://github.com/exercism/problem-specifications/blob/main/exercises/largest-series-product/description.md) (5) // 79a9cf08
- [x] [roman-numerals](https://github.com/exercism/problem-specifications/blob/main/exercises/roman-numerals/description.md) (5) // ffd8763
- [x] [resistor-color-trio](https://github.com/exercism/problem-specifications/blob/main/exercises/resistor-color-trio/description.md) (6) // #46
- [x] [sgf-parsing](https://github.com/exercism/problem-specifications/blob/main/exercises/sgf-parsing/description.md) (6) // #62
- [x] [high-scores](https://github.com/exercism/problem-specifications/blob/main/exercises/high-scores/description.md) (7) // 009679130
- [x] [run-length-encoding](https://github.com/exercism/problem-specifications/blob/main/exercises/run-length-encoding/description.md) (7) // a836243b3
- [x] [simple-linked-list](https://github.com/exercism/problem-specifications/blob/main/exercises/simple-linked-list/description.md) (7) // ccd2b53443
- [x] [bowling](https://github.com/exercism/problem-specifications/blob/main/exercises/bowling/description.md) (8) // 3b68985bc
- [x] [pov](https://github.com/exercism/problem-specifications/blob/main/exercises/pov/description.md) (8) // 033a49d6f4
- [x] [circular-buffer](https://github.com/exercism/problem-specifications/blob/main/exercises/circular-buffer/description.md) (9) // 097916d

### instructions:

1. Comment here to let anybody know which exercises are being worked on,
2. Clone this repo,
3. Run exercise generator:
```shell
$ red _tools/generate-practice-exercise.red <exercise-slug>
```
4. In `exercises/practice/<exercise-slug>/<exercise-slug>-test.red` change comments like this, to test example solution:
```red
; test-init/limit %exercise-slug.red 1
test-init/limit %.meta/example.red 1
```
5. Solve exercise example, by editing `exercises/practice/<exercise-slug>/.meta/example.red`,
6. Run tests. You'll need to change second argument of `test-init` function from `1` to how many tests you want to run in `<exercise-slug>-test.red`.
```shell
$ cd exercises/practice/<exercise-slug>
$ red <exercise-slug>-test.red
```
7. Once your solution passes all the tests, remember to revert changes in `test-init` line: uncomment solution file, comment example file and change `limit` to `1` (second argument).
8. Change exercise's difficulty in track's `config,json`. If you want, add `practices` and `prerequisites` concepts. Copy exercise's config to proper position, so that all exercises are sorted from easiest to toughest.
9. Make a commit to separate branch and make Pull Requeset.

**Comment by loziniak** [comment]
checklist:
- [x] hello world
- [x] two-fer
- [ ] reverse string

**Comment by wallysilva** [comment]
I will do the hello-world (1)

**Comment by loziniak** [comment]
It already has a solution, but of course you can solve it as a learning example. Most needed are solution for exercises with empty checkbox.

**Comment by wallysilva** [comment]
I believe you can find some of the solutions in the Rosetta Code. For instance, the solution for the Roman Numerals problem: http://www.rosettacode.org/wiki/Roman_numerals/Decode#Red

**Comment by loziniak** [comment]
@dander exactly. This is a workflow in every Exercism track. Although nothing can stop you from running all the tests from the very beginning. I suspect it's a sort of TDD good practice perhaps. BTW nice to see you involved!

**Comment by loziniak** [comment]
> I've started working on *sgf-parsing*

How is it going? Perhaps we could go live soon, do you want to finish that? If not, I'll take it.

**Comment by dander** [comment]
I had to stop for a while because of general life stuff going on, but I've been trying to get back to it lately. I was finding it a bit tricky to indicate in the tests what was wrong with the outputs in a clear way. Also, trying to figure out an appropriate way to handle expected errors. Have you encountered that in some of the other challenges?

**Comment by loziniak** [comment]
Yes, this **error** pattern appeared *largest-series-product* for me: 79a9cf08ca76 . I extended a testing "framework" lately (009679130bd87f5cfb27f5240013124bc94a95cc), errors should still work. You should be able to `throw` a map with *error* key, or `cause-error` with appropriate message.

To satisfy structured output expectations, I would suggest you just return a map from tested function.

**Comment by dander** [comment]
I pushed up an example solution for the `sgf-parsing` exercise (finally). #62

I initially wanted to use parse with collect/keep, but since the outputs expect nested maps, and collect only can generate blocks, I wasn't sure how that would work. So instead I used a stack to keep track of the current location in the data structure to insert child nodes. I found this problem to be quite difficult to get right. I'm not sure that using `parse` is the easiest solution for it, but it seems like a natural place to display the feature of the language.

I'm looking into adding the project metadata pieces. I am considering adding concepts for `parse`, and `recursion`. Is there a catalog of existing concepts somewhere that I should reference? Is there anything I need to know about the uuids, or do I just generate a new one?
